### ISSUE TITLE
write_dta fails with "variable_value_labels type must be boolean or be 1 or 0" if only NaNs and missing_user_values in series

### ISSUE THREAD

**Issue by eirki** [Roche/pyreadstat#158]
**Describe the issue**
When saving a dataframe as DTA, if a column contains only NaNs and `missing_user_values`, it can be incorrectly identified as a boolean variable (I think). If  `variable_value_labels` for other valid values are specified, this causes an exception, since boolean variables can only have value labels for boolean, 1 or 0.

**To Reproduce**

```python
import numpy as np
import pandas as pd
import pyreadstat

pyreadstat.write_dta(
    pd.DataFrame({"var": [np.nan, "a", "b"]}),
    "./myfile.dta",
    variable_value_labels={
        "var": {
            1: "Val 1",
            2: "Val 2",
            3: "Val 3",
            "a": "Missing A",
            "b": "Missing B",
        }
    },
    missing_user_values={"var": ["a", "b"]},
    version=14  # can be any version
)
```
Exception:
```
Traceback (most recent call last):
    pyreadstat.write_dta(
  File "pyreadstat/pyreadstat.pyx", line 835, in pyreadstat.pyreadstat.write_dta
  File "pyreadstat/_readstat_writer.pyx", line 794, in pyreadstat._readstat_writer.run_write
  File "pyreadstat/_readstat_writer.pyx", line 698, in pyreadstat._readstat_writer.run_write
  File "pyreadstat/_readstat_writer.pyx", line 334, in pyreadstat._readstat_writer.set_value_label
pyreadstat._readstat_parser.PyreadstatError: variable_value_labels: type of Value 2 in variable var must be boolean or be 1 or 0
```

**System**
pandas==1.2.4
pyreadstat==1.1.4
OS: Ubuntu 20.04

_(Edit: Sorry for all the issues I've been piling on you lately! I have been using this library to great effect, and am really grateful for all the work you're putting in to it.)_

**Comment by ofajardo** [comment]
solved on dev.

**Comment by ofajardo** [comment]
solved in version 1.1.5
